### PR TITLE
[SPARK-7645][Streaming][WebUI]Show milliseconds in the UI if the batch interval < 1 second

### PR DESCRIPTION
**Pull request by zsxwing**
I also updated the summary of the Streaming page.

screen shot 2015-05-14 at 11 52 59 am
screen shot 2015-05-14 at 12 39 33 pm

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @tdas

File: streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala [hunk 1]
```diff
@@ -288,7 +290,14 @@ private[ui] class BatchPage(parent: StreamingTab) extends WebUIPage("batch") {
...
       throw new IllegalArgumentException(s"Missing id parameter")
     }
-    val formattedBatchTime = SparkUIUtils.formatDate(batchTime.milliseconds)
+    val batchTimeFormat =
+      if (streamingListener.batchDuration < 1000) {
+        new SimpleDateFormat("yyyy/MM/dd HH:mm:ss.SSS")
+      } else {
+        // If batchInterval >= 1 second, don't show milliseconds
+        new SimpleDateFormat("yyyy/MM/dd HH:mm:ss")
+      }
```
Comment (andrewor14) [inline streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala]: these few lines are duplicated in a few places. Can you put it in `streaming.ui.UIUtils`? I will also find this useful in my code.

**Comment by andrewor14** [comment]
One small comment LGTM otherwise.

File: streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala (hunk 1 above)
Comment (zsxwing) [inline streaming/src/main/scala/org/apache/spark/streaming/ui/BatchPage.scala]: OK. I will update it.

**Comment by andrewor14** [comment]
Great, merging into master 1.4.

**Pull request closed without merging**